jupyterhubutils/lsstmgr/quotamanager.py

```diff
@@ -0,0 +1,153 @@
+'''Quota support for LSST LSP Jupyterlab and Dask pods.
+'''
+import json
+import os
+from kubernetes.client import V1ResourceQuotaSpec
+from kubernetes.client.rest import ApiException
+from kubernetes import client
+
+from ..utils import make_logger
+
+
+class LSSTQuotaManager(object):
+    quota = {}
+    _custom_resources = {}
+    resourcemap = None
+
+    def __init__(self, *args, **kwargs):
+        self.log = make_logger()
+        self.log.debug("Creating LSSTQuotaManager")
+        self.parent = kwargs.pop('parent')
+
+    def read_resource_map(self, resource_file=None):
+        rfile = "/opt/lsst/software/jupyterhub/resources/resourcemap.json"
+        if not resource_file:
+            resource_file = rfile
+        if not os.path.exists(resource_file):
+            nf_msg = ("Could not find resource definition file" +
+                      " at '{}'".format(resource_file))
+            self.log.error(nf_msg)
+            return None
+        with open(resource_file, "r") as rf:
+            resmap = json.load(rf)
+        return resmap
+
+    def set_custom_user_resources(self):
+        '''Create custom resource definitions for user.
+        '''
+        if not self.resourcemap:
+            self.log.info("No resource map found; generating.")
+            self.resourcemap = self.read_resource_map()
+            if not self.resourcemap:
+                return
```

Comment:
You only call read_resource_map() here, so just call it in init at the start and be done with it